src/ngx_http_lua_kong_module.c

```diff
@@ -309,6 +311,52 @@ ngx_http_lua_kong_ffi_get_full_client_certificate_chain(ngx_http_request_t *r,
 }
 
 
+int
+ngx_http_lua_kong_ffi_set_grpc_authority(ngx_http_request_t *r,
+    const char *buf, size_t buf_len)
+{
+    u_char                      *host;
+    ngx_http_lua_kong_ctx_t     *ctx;
+
+    ctx = ngx_http_lua_kong_get_module_ctx(r);
+    if (ctx == NULL) {
+        return NGX_ERROR;
+    }
+
+    host = ngx_palloc(r->pool, buf_len);
```

Comment:
we can skip alloc and just clear `ctx->grpc_authority` if user called `set_authority("")`

Reply:
This was discussed with @fffonion offline, and we instead throw an error when people attempts to set empty header because it won't work correctly.